src/main/java/games/strategy/engine/message/RemoteInterfaceHelper.java

```diff
@@ -19,41 +17,26 @@ static int getNumber(final String methodName, final Class<?>[] argTypes, final C
         .orElseThrow(() -> new IllegalStateException("Method not found: " + methodName));
   }
 
-  static Tuple<String, Class<?>[]> getMethodInfo(final int methodNumber, final Class<?> remoteInterface) {
+  static Method getMethod(final int methodNumber, final Class<?> remoteInterface) {
     final Method[] methods = remoteInterface.getMethods();
     Arrays.sort(methods, methodComparator);
-    return Tuple.of(methods[methodNumber].getName(), methods[methodNumber].getParameterTypes());
+    return methods[methodNumber];
   }
 
   /**
    * get methods does not guarantee an order, so sort.
    */
-  private static final Comparator<Method> methodComparator = (o1, o2) -> {
-    if (o1 == o2) {
-      return 0;
-    }
-    if (!o1.getName().equals(o2.getName())) {
-      return o1.getName().compareTo(o2.getName());
-    }
-    final Class<?>[] t1 = o1.getParameterTypes();
-    final Class<?>[] t2 = o2.getParameterTypes();
-    if ((t1 == null) && (t2 == null)) {
-      return 0;
-    }
-    if (t1 == null) {
-      return -1;
-    }
-    if (t2 == null) {
-      return 1;
-    }
-    if (t1.length != t2.length) {
-      return t1.length - t2.length;
-    }
-    for (int i = 0; i < t1.length; i++) {
-      if (!t1[i].getName().equals(t2[i].getName())) {
-        return t1[i].getName().compareTo(t2[i].getName());
-      }
-    }
-    return 0;
-  };
+  private static final Comparator<Method> methodComparator = Comparator
+      .comparing(Method::getName)
+      .thenComparing(Method::getParameterTypes,
+          Comparator.comparingInt((Class<?>[] a) -> a.length)
+              .thenComparing((o1, o2) -> {
```

Comment:
Note for the future: In java 9 this code can be replaced with https://docs.oracle.com/javase/9/docs/api/java/util/Arrays.html#compare-T:A-T:A-java.util.Comparator-
e.g.:
```java
private static final Comparator<Method> methodComparator = Comparator
  .comparing(Method::getName)
  .thenComparing(Method::getParameterTypes,
    (o1, o2) -> Arrays.compare(o1, o2, Comparator.comparing(Class::getName)));
```